Feat/doc build (#51051)
qgis/QGIS: INSTALL.md

ninja qgis_desktop
```

### 3.7.1 Available compilation flags

> QGIS built is tunable according to your needs. Many flags are available to activate or deactivate some functionalities:

Reviewer: ```suggestion
QGIS build is tunable according to your needs. Many flags are available to activate or deactivate some functionalities:
```

Author: done!

* `WITH_3D`: Determines whether QGIS 3D library should be built
* `WITH_ANALYSIS`: Determines whether QGIS analysis library should be built
* `WITH_APIDOC`: Determines whether the QGIS API doxygen documentation should be built
* `WITH_ASAN`: Use AddressSanitizer

> * `WITH_ASTYLE`: If you plan to contribute you should reindent with scripts/prepare_commit.sh (using 'our' astyle)

Reviewer 2: This isn't really necessary anymore -- eg I don't build WITH_ASTYLE enabled. It was needed years ago when distros had older astyle which didn't support some of the code we used, but could likely be safely removed now.

Author: done

* `WITH_AUTH`: Determines whether QGIS authentication methods should be built

> * `WITH_BINDINGS`: Determines whether python bindings should be built

Reviewer 2: ```suggestion
* `WITH_BINDINGS`: Determines whether Python bindings should be built
```

Author: I have changed all the python by Python

* `WITH_CLANG_TIDY`: Use Clang tidy
* `WITH_CLAZY`: Determines whether Clazy checks should be enabled
* `WITH_COPC`: Determines whether Cloud Optimized Point Cloud (COPC) support should be built
* `WITH_CORE`: Determines whether QGIS core should be built.
* `WITH_CRASH_HANDLER`: Determines whether the QGIS crash handler application should be built
* `WITH_CUSTOM_WIDGETS`: Determines whether QGIS custom widgets for Qt Designer should be built
* `WITH_DESKTOP`: Determines whether QGIS desktop should be built
* `WITH_EPT`: Determines whether Entwine Point Cloud (EPT) support should be built
* `WITH_GRASS${GRASS_CACHE_VERSION}`: Determines whether GRASS ${GRASS_SEARCH_VERSION} plugin should be built
* `WITH_GUI`: Determines whether QGIS GUI library (and everything built on top of it) should be built
* `WITH_HANA`: Determines whether SAP HANA Spatial support should be built
* `WITH_INTERNAL_${ulib}`: Determines whether python ${lib} should be included

> * `WITH_INTERNAL_LAZPERF`: Determines whether LazPerf should be built from internal copy

Reviewer 2: ```suggestion
* `WITH_INTERNAL_LAZPERF`: Determines whether LazPerf should be built from internal copy (recommended)
```

Author: Does not exist anymore. Bad `find` request.

Author: Bad day today... I did not rebase on the right branch!

> * `WITH_INTERNAL_MDAL`: Determines whether MDAL should be built from internal copy

Reviewer 2: ```suggestion
* `WITH_INTERNAL_MDAL`: Determines whether MDAL should be built from internal copy (recommended)
```

Author: done

* `WITH_INTERNAL_POLY2TRI`: Determines whether POLY2TRI should be built from internal copy
* `WITH_INTERNAL_QWTPOLAR`: Use internal build of QwtPolar

> * `WITH_OAUTH2_PLUGIN`: Build OAuth2 authentication method plugin

Reviewer: ```suggestion
* `WITH_OAUTH2_PLUGIN`: Determines whether OAuth2 authentication method plugin should be built
```

Homogenization

Author: I am agree but this text is extracted from CMakeLists.txt, should I apply your suggestion?

Reviewer: Based on Nyall's review, I would say you are free to change it if you want ! 😀

Author: I will modified the INSTALL.md and CMakeLists.txt accordingly

* `WITH_ORACLE`: Determines whether Oracle support should be built
* `WITH_PDAL`: Determines whether PDAL support should be built
* `WITH_POSTGRESQL`: Determines whether POSTGRESQL support should be built

> * `WITH_PY_COMPILE`: Determines whether Python modules in staged or installed locations are byte-compiled

Reviewer: ```suggestion
* `WITH_PY_COMPILE`: Determines whether Python modules in staged or installed locations should be byte-compiled
```

Not sure

Author: done

* `WITH_QGIS_PROCESS`: Determines whether the standalone \"qgis_process\" tool should be built
* `WITH_QSCIAPI`: Whether to generate PyQGIS QScintilla2 API file. (For devs) run 'make qsci-pap-src' in between QGIS build and install to regenerate .pap file in source tree for console auto-completion.
* `WITH_QSPATIALITE`: Determines whether QSPATIALITE sql driver should be built
* `WITH_QT5SERIALPORT`: Determines whether Qt5SerialPort should be tried for GPS positioning

> * `WITH_QT6`: Enable (broken, experimental) Qt6 support

Reviewer 2: ```suggestion
* `WITH_QT6`: Enable (experimental) Qt6 support
```

it's not broken :)

Author: done

* `WITH_QTWEBKIT`: Enable QtWebkit support
* `WITH_QUICK`: Determines whether QGIS Quick library should be built

> * `WITH_QWTPOLAR`: Determines whether QwtPolar should be built

Reviewer 2: ```suggestion
* `WITH_QWTPOLAR`: Determines whether QwtPolar is available and or whether functionality requiring QwtPolar should be disabled
```

Author: done
